unittests/data/type_traits.hpp

```diff
@@ -544,13 +586,14 @@ namespace yes{
     struct x{
         x(){}
     };
+#ifdef __castxml__
+    typedef details::const_item const_item;
+    typedef details::const_container const_container;
+#endif
```

Comment:
The difference in behavior between `std::is_trivially_constructible` and `boost::has_trivial_constructor` is a Boost bug. The example linked to above uses Boost 1.59 and that [passes](http://melpon.org/wandbox/permlink/FFpYgWyV2gv4TdYT), but it [fails](http://melpon.org/wandbox/permlink/lcEfAhK3QAT9eCjX) on Boost 1.60.

The next step is to look into what's going wrong with the `has_trivial_constructor` trait when using CastXML.